pkg/imgpkg/util/tag.go

```diff
@@ -28,3 +30,20 @@ func BuildDefaultUploadTagRef(item WithDigest, importRepo regname.Repository) (r
 	}
 	return uploadTagRef, nil
 }
+
+// BuildLegibleUploadTagRef Builds a legible tag from original image reference
+func BuildLegibleUploadTagRef(itemRef string, importRepo regname.Repository) (regname.Tag, error) {
+	// ex.:
+	// 1. registry.f.q.d.n/myorg/my-img@sha256:012345679e2a06317dc02abcde123759eb9e6105a0731224a0c63898abcde123
+	// 2. registry.f.q.d.n/myorg/my-img:some-tag@sha256:012345679e2a06317dc02abcde123759eb9e6105a0731224a0c63898abcde123
+	origRepoAndTag := strings.Split(itemRef, "@")
```

Comment:
This can be accomplished with 
```go
digest, err := regname.NewDigest(itemRef)

origRepoPath := a.RepositoryStr()
dashedRepo := strings.ReplaceAll(origRepoPath, "/", "-")
tag := fmt.Sprintf("%s-%s-%s.imgpkg", dashedRepo[len(dashedRepo)-49:], digest.Algorithm, digest.Hex)
```

I did not try to compile this, so please take it with a grain of salt, also there is the problem of `dashedRepo` being smaller than 49